### PR TITLE
Document/Refer Function feature more properly

### PR DESCRIPTION
**Pull request by feeblefakie**
This PR fixes the following issues around `Function` based on feedbacks:
* The Function guide is not properly referred from another guide.
* The Function guide is not easy to follow, especially in how to execute part.

### CODE REVIEW

File: docs/how-to-write-function.md [hunk 1]
```diff
@@ -67,7 +67,13 @@ Similar to a Contract using `Ledger` object to manage assets, a Function uses `D
 
 ### How Functions and Contracts are tied together
 
-Functions to be executed are specified in a contract argument with `__functions__` JSON key. As described in [the contract guide](how-to-write-contract.md#write-a-complex-contract), a contract can invoke another contract, so multiple contracts and multiple functions can be grouped together.
+Functions to be executed are specified in a contract argument in a JSON format with `__functions__` as a key and an array of function IDs as a value as follows:
```
Comment (jnmt) [inline docs/how-to-write-function.md:70]: I don't fully know this page is the best place to write details about the function, but developers might want to know the registration of functions and correspondence of params (e.g., function ID/name) through the same example on this page. For example, how about adding the following?

---

First functions must be registered to execute with contracts as follows:
```
register-function --properties client.properties --function-id test-function --function-binary-name com.example.function.PaymentFunction --function-class-file PaymentFunction.class
```

File: docs/how-to-write-function.md [hunk 2]
```diff
@@ -67,7 +67,13 @@ Similar to a Contract using `Ledger` object to manage assets, a Function uses `D
...
+
+```
+execute-contract --properties client.properties --contract-id test-contract --contract-argument '{..., "__functions__": ["test-function"]}'
```
Comment (jnmt) [inline docs/how-to-write-function.md:73]: Is it natural we have ``--function-argument '{...}'`` if using the function?

**Review by jnmt — COMMENTED** [review]
LGTM, but I left few comments as a beginner of the function user : ) Thanks.

File: docs/how-to-write-contract.md [hunk 3]
```diff
@@ -89,6 +89,14 @@ One common way of creating a non-deterministic contract is to generate the time
 Such a contract will produce different outputs each time it is executed, and makes the system unable to detect tampering.
 If you need to use the time in a contract, you should pass it to the contract as an argument.
 
+### Deleting an asset
+
+The assets registered thorough contracts are not able to be deleted to provide tamper-evidence.
```
Comment (supl) [inline docs/how-to-write-contract.md:94]: Is `thorough` a typo of `through`?

**Review by supl — APPROVED** [review]
LGTM.
I marked a possible typo for your reference.
Thank you.

File: docs/how-to-write-contract.md (hunk 3 above)
Comment (feeblefakie) [reply docs/how-to-write-contract.md:94]: Yes, thank you!

File: docs/how-to-write-function.md (hunk 1 above)
Comment (feeblefakie) [reply docs/how-to-write-function.md:70]: Thank you!
As pointed out, there is not really "how to use" guide anywhere, so let me add something like that including how to register.

File: docs/how-to-write-function.md (hunk 2 above)
Comment (feeblefakie) [reply docs/how-to-write-function.md:73]: True. Let me update this.
Thank you!

**Comment by feeblefakie** [comment]
@jnmt It's updated in #26e66c0. PTAL when you get another chance!

File: docs/how-to-write-function.md [hunk 4]
```diff
@@ -65,17 +65,35 @@ Similar to a Contract using `Ledger` object to manage assets, a Function uses `D
...
+
+You then need to register a Function to Ledger before used like you register a Contract.
+
+```
+register-function --properties client.properties --function-id test-function --function-binary-name com.example.function.TestFunction --function-class-file /path/to/TestFunction.class
+```
+
+#### Execute a registered Function
+
+Functions to be executed are specified in a contract argument in a JSON format with `__functions__` as a key and an array of function IDs as a value as follows:
```
Comment (feeblefakie) [inline docs/how-to-write-function.md:88]:
```suggestion
Functions to be executed are specified in a contract argument in a JSON format with `_functions_` as a key and an array of function IDs as a value as follows:
```

File: docs/how-to-write-function.md [hunk 5]
```diff
@@ -65,17 +65,35 @@ Similar to a Contract using `Ledger` object to manage assets, a Function uses `D
...
+
+```
+execute-contract --properties client.properties --contract-id test-contract --contract-argument '{..., "__functions__": ["test-function"]}' --function-argument '{...}'
```
Comment (feeblefakie) [inline docs/how-to-write-function.md:91]:
```suggestion
execute-contract --properties client.properties --contract-id test-contract --contract-argument '{..., "_functions_": ["test-function"]}' --function-argument '{...}'
```

**Review by ypeckstadt — COMMENTED** [review]
LGTM. Added a few very minor nits for grammar.

File: docs/how-to-write-function.md [hunk 6]
```diff
@@ -65,17 +65,35 @@ Similar to a Contract using `Ledger` object to manage assets, a Function uses `D
...
+
+You then need to register a Function to Ledger before used like you register a Contract.
+
+```
+register-function --properties client.properties --function-id test-function --function-binary-name com.example.function.TestFunction --function-class-file /path/to/TestFunction.class
+```
+
+#### Execute a registered Function
+
+Functions to be executed are specified in a contract argument in a JSON format with `_functions_` as a key and an array of function IDs as a value as follows:
```
[the remark this answers is not in the document]
Comment (feeblefakie) [reply docs/how-to-write-function.md:88]: Thanks!
I think the suggested sentence seems to have a bit long concatenated noun, so how does the following sound to you?
```
Functions that are being executed are specified in a contract argument ...
```

**Review by feeblefakie — COMMENTED** [review]
@ypeckstadt Thank you for the comments!
For the most part, I agree and I just applied the suggestions.
I left an alternative sentence for one suggestion. PTAL when you get a chance.

**Comment by feeblefakie** [comment]
@ypeckstadt I'll merge it. Please create a PR if you feel the doc still needs a revision. :bow:

**Pull request merged**